packages/x-components/src/x-modules/tagging/service/__tests__/session.service.spec.ts

```diff
@@ -0,0 +1,50 @@
+import { DefaultSessionService } from '../session.service';
+import { MockedStorageService } from './utils/mock-storage.service';
+
+describe('testing session id service', () => {
+  const prefix = 'test';
+  const mockedStorageService = new MockedStorageService(prefix);
+  const sessionService = new DefaultSessionService(mockedStorageService, 1);
+
+  afterEach(() => {
+    jest.clearAllMocks();
+  });
+
+  it('creates a new session id', () => {
+    const session = sessionService.getSessionId();
+    expect(mockedStorageService.getPrefix()).toBe(prefix);
+    expect(mockedStorageService.getSessionIdSpy).toHaveBeenCalledTimes(1);
+    expect(mockedStorageService.setSessionIdSpy).toHaveBeenCalledTimes(1);
+    expect(mockedStorageService.removeSessionIdSpy).not.toHaveBeenCalled();
```

Comment:
I don't think this check is necessary here. Why should we check `removeSessionId` if has been called or not here?

```suggestion
```